### PR TITLE
ES6 Modules

### PR DESCRIPTION
**Pull request by daudi13**
## In this Project:
- I used ES6 modules
- I separated different functions of my code in different modules
- I applies javascript best practices
- Installed and used Luxon as an external library

### CODE REVIEW

**Comment by daudi13** [comment]
All my linter errors are being generated by the luxon library. Is it allowed to change the code in an external libraby?

**Comment by TedLivist** [comment]
Hey @daudi13, you can create an **`.eslintignore`** file and ignore the `modules/luxon` file path. You can refer to this [link](https://eslint.org/docs/user-guide/configuring/ignoring-code#the-eslintignore-file) for further reading.

On the other hand, there are other linter errors not generated from luxon, you can fix them too

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: INVALID CODE REVIEW REQUEST 🛑 🛑

Hi @daudi13,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### You have submitted a project that has failing linters and contains the node_modules

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: modules/Class/mainClass.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+  removeBtn.innerText = 'remove';
+
+  removeBtn.onclick = () => {
+    UpdateDisplay.delBook(bookItem, pos);
+  };
+
+  bookBlock.appendChild(removeBtn);
+  UpdateDisplay.listSection.appendChild(bookBlock);
+}
```
Comment (TedLivist) [inline modules/Class/mainClass.js]: - [ ] For better code readability, I'd recommend you indent your code properly.

File: modules/time/time.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { timebox } from '../selector/selectors.js';
+import { DateTime } from '../luxon/src/luxon.js';
+
+export function time() {
```
Comment (TedLivist) [inline modules/time/time.js]: - [ ] Kindly refactor this function to use ES6 syntax as it makes your code look neater and takes less space.

File: modules/Class/mainClass.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+static formBtn = selectors.formBtn;
+
+static bookAuthor = selectors.bookAuthor;
+
+static listBtn = selectors.listBtn;
+
+static addActive = selectors.addActive;
+
+// create new book
+static addBooks() {
```
Comment (TedLivist) [inline modules/Class/mainClass.js]: - [ ] Kindly refactor to arrow function

File: modules/Class/mainClass.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  );
+  UpdateDisplay.books.push(bookItem);
+  localStorage.setItem('books', JSON.stringify(UpdateDisplay.books));
+
+  UpdateDisplay.bookAuthor.value = '';
+  UpdateDisplay.bookTitle.value = '';
+  UpdateDisplay.addBookItem(bookItem, UpdateDisplay.books.length - 1);
+}
+
+static delBook(bookItem, pos) {
```
Comment (TedLivist) [inline modules/Class/mainClass.js]: - [ ] Kindly refactor to arrow function

File: modules/Class/mainClass.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  const bookBlock = document.getElementById(pos);
+  UpdateDisplay.books = UpdateDisplay.books.filter((item) => item !== bookItem);
+  localStorage.setItem('books', JSON.stringify(UpdateDisplay.books));
+  UpdateDisplay.listSection.removeChild(bookBlock);
+}
+
+static updateUi() {
```
Comment (TedLivist) [inline modules/Class/mainClass.js]: - [ ] Change to arrow function

File: modules/Class/mainClass.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    UpdateDisplay.books.forEach((bookItem, pos) => {
+      UpdateDisplay.addBookItem(bookItem, pos);
+    });
+  } else {
+    localStorage.setItem('books', '');
+    UpdateDisplay.books = [];
+  }
+}
+
+static addBookItem(bookItem, pos) {
```
Comment (TedLivist) [inline modules/Class/mainClass.js]: - [ ] Kindly change to arrow function

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @daudi13,

Good job so far!
✔️ Good documentation
✔️ Books can be saved and removed
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
#### _Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by dasileker — APPROVED** [review]
## **Hi @daudi13,** 

## **Congratulations! 🎉 Project Approved ✅**
Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 
 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**